### PR TITLE
bci-001/Basic layout & logic

### PR DESCRIPTION
**Pull request by Myxjiew**
Created basic layout;
Created MaterialUI form and validation;
Divided into components;

### CODE REVIEW

File: src/app/shared/components/header/header.component.sass [hunk 1]
```diff
@@ -0,0 +1,34 @@
+.header
```
Comment (MMaksymivUa) [inline src/app/shared/components/header/header.component.sass:1]: SASS????????? Do not use sass, you should start working with scss

File: src/app/my-blog/components/layout/layout.component.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import { Subject, takeUntil } from 'rxjs';
+import { IPost } from '@shared/interfaces/blog-item.interface';
+import { BlogService } from '@shared/services/blog.service';
+
+@Component({
+  selector: 'app-layout',
+  templateUrl: './layout.component.html',
+  styleUrls: ['./layout.component.sass'],
+})
+export class LayoutComponent implements OnInit, OnDestroy {
```
Comment (MMaksymivUa) [inline src/app/my-blog/components/layout/layout.component.ts]: create post-list and use it inside layout

File: src/app/my-blog/components/layout/layout.component.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<main class="layout">
+  <div class="layout__blog-list">
+    <ul>
+      <ng-container *ngFor="let post of posts">
```
Comment (MMaksymivUa) [inline src/app/my-blog/components/layout/layout.component.html]: <app-blog-item *ngFor="let post of posts" [post]="post"></app-blog-item>
No need of ng-container

File: src/app/my-blog/components/blog-item/blog-item.component.html [hunk 4]
```diff
@@ -0,0 +1,34 @@
+<li>
```
Comment (MMaksymivUa) [inline src/app/my-blog/components/blog-item/blog-item.component.html]: bad HTML semantic, do not use list items

File: src/app/my-blog/components/blog-item/blog-item.component.html [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  <div class="post">
```
Comment (MMaksymivUa) [inline src/app/my-blog/components/blog-item/blog-item.component.html]: just check if the post exists

File: src/app/my-blog/components/aside-tags/aside-tags.component.sass [hunk 6]
```diff
@@ -0,0 +1,9 @@
+.aside
+    background: #f2f2f2
+    padding: 8px
+
+    &__label
+        margin-bottom: 12px
+
+
```
Comment (MMaksymivUa) [inline src/app/my-blog/components/aside-tags/aside-tags.component.sass]: formatting

Comment (Myxjiew) [reply src/app/my-blog/components/aside-tags/aside-tags.component.sass]: fixed

File: src/app/my-blog/components/blog-item/blog-item.component.html (hunk 5 above)
Comment (Myxjiew) [reply src/app/my-blog/components/blog-item/blog-item.component.html]: fixed

File: src/app/my-blog/components/layout/layout.component.html (hunk 3 above)
Comment (Myxjiew) [reply src/app/my-blog/components/layout/layout.component.html]: fixed

File: src/app/my-blog/components/blog-item/blog-item.component.html (hunk 4 above)
Comment (Myxjiew) [reply src/app/my-blog/components/blog-item/blog-item.component.html]: fixed

File: src/app/my-blog/components/layout/layout.component.ts (hunk 2 above)
Comment (Myxjiew) [reply src/app/my-blog/components/layout/layout.component.ts]: fixed

**Pull request merged**